packages/ionic/test/number-control.spec.ts

```diff
@@ -0,0 +1,221 @@
+        expect(
+            numberControlTester(
+                uischema,
+                {
+                    type: 'object',
+                    properties: {
+                        foo: { type: 'integer' }
+                    }
+                }
+            )
+        ).toBe(2);
+    });
+});
+
+describe(
+    'Ionic number control', () => {
+        let fixture;
+
+        beforeEach(() => {
+            TestBed.configureTestingModule({
+                declarations: [NumberControlRenderer],
+                imports: [IonicModule.forRoot(NumberControlRenderer)],
+                providers: [
+                    { provide: Platform, useClass: PlatformMock },
+                    { provide: NgRedux, useFactory: MockNgRedux.getInstance }
+                ]
+            }).compileComponents();
+
+            MockNgRedux.reset();
+            fixture = TestBed.createComponent(NumberControlRenderer);
+        });
+
+        it('should render floats', () => {
+            renderFloat(fixture, defaultTestData, () => {
+                const textInput: DebugElement = fixture.debugElement.query(By.directive(TextInput));
+                const textInputInstance = textInput.componentInstance;
+                expect(textInputInstance.value).toBe(123.123);
+                expect(textInputInstance.step).toBe(0.1);
+                expect(textInputInstance.disabled).toBe(false);
+            });
+        });
+
+        it('should render integer', () => {
+            renderInteger(
+                fixture,
+                {
+                    data: { foo: 123 },
+                    schema: {
+                        type: 'object',
+                        properties: {
+                            foo: { type: 'integer' }
+                        }
+                    },
+                    uischema: defaultTestData.uischema
+                },
+                () => {
+                    const textInput = fixture.debugElement.query(By.directive(TextInput));
+                    const textInputInstance = textInput.componentInstance;
+                    expect(textInputInstance.value).toBe(123);
+                    expect(textInputInstance.step).toBe(1);
+                    expect(textInputInstance.disabled).toBe(false);
+                });
+        });
+
+        it('should support updating the state', () => {
+            updateFloatState(fixture, defaultTestData, () => {
+                const textInput = fixture.debugElement.query(By.directive(TextInput));
+                const textInputInstance = textInput.componentInstance;
+                expect(textInputInstance.value).toBe(456.456);
+            });
+        });
+
+        // possibly related to https://github.com/ionic-team/ionic/issues/14370
+        xit('should support updating with undefined', () => {
+            updateWithUndefined(fixture, defaultTestData, () => {
+                const textInput = fixture.debugElement.query(By.directive(TextInput));
```

Comment:
`const textInput = fixture.debugElement.query(By.directive(TextInput));
                const textInputInstance = textInput.componentInstance;`
could be extracted